src/request_collector/server.py

```diff
@@ -56,25 +128,122 @@ def _run(self):
             self.task_list.remove(task)
 
     def stop(self):
-        self.transport.stop()
         self.stop_event.set()
 
-    def on_message_event(self, message):
-        """This handles messages received over the Transport"""
-        log.debug(message)
-        if isinstance(message, MonitorRequest):
+    def _handle_message(self, room, event) -> bool:
+        """ Handle text messages sent to listening rooms """
+        if (
+                event['type'] != 'm.room.message' or
+                event['content']['msgtype'] != 'm.text' or
+                self.stop_event.ready()
+        ):
+            # Ignore non-messages and non-text messages
+            return False
+
+        sender_id = event['sender']
+
+        user = self.client.get_user(sender_id)
```

Comment:
I'd suggest implementing [_some_ caching](https://github.com/raiden-network/raiden/blob/1f54f8cc85d6b27029184cb602b5cd2c2395a88c/raiden/network/transport/matrix/transport.py#L1160-L1176) on this, as `client.get_user` is just a factory function, always creating new instances of `User` with empty displayname (and therefore, needing re-fetching it inside `validate_userid_signature`). `validate_userid_signature` caches the `recover` call, but needs both `user_id` and `displayname` to be correctly set, or it'll fetch it on each call. The complete user list (including displaynames) for a room comes on initial sync, so you may just get keep them in sync on `monitoring_room._members`, as done in raiden, or other caching methods you may find useful. Or not, and let it be ;)

Reply:
Thanks for the hint, I'll do this in a follow-up PR.